Code/Tools/ProjectManager/Source/ProjectExportWorker.cpp

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ProjectExportWorker.h>
+#include <ProjectManagerDefs.h>
+#include <PythonBindingsInterface.h>
+#include <ProjectUtils.h>
+
+#include <QDir>
+#include <QFile>
+#include <QProcess>
+#include <QProcessEnvironment>
+#include <QTextStream>
+#include <QThread>
+
+namespace O3DE::ProjectManager
+{
+    ProjectExportWorker::ProjectExportWorker(const ProjectInfo& projectInfo)
+        : QObject()
+        , m_projectInfo(projectInfo)
+    {
+    }
+
+    void ProjectExportWorker::ExportProject()
+    {
+        auto result = ExportProjectForPlatform();
+
+        if (result.IsSuccess())
+        {
+            emit Done();
+        }
+        else
+        {
+            emit Done(result.GetError());
+        }
+    }
+
+    QString ProjectExportWorker::GetLogFilePath() const
+    {
+        QDir logFilePath(m_projectInfo.m_path);
+        // Make directories if they aren't on disk
+        if (!logFilePath.cd(ProjectBuildPathPostfix))
+        {
+            bool madeDirectory = logFilePath.mkpath(ProjectBuildPathPostfix);
+            if (!madeDirectory)
+            {
+                return QString("");
+            }
+            logFilePath.cd(ProjectBuildPathPostfix);
+        }
+        if (!logFilePath.cd(ProjectBuildPathCmakeFiles))
+        {
+            bool madeDirectory = logFilePath.mkpath(ProjectBuildPathCmakeFiles);
+            if (!madeDirectory)
+            {
+                return QString("");
+            }
+            logFilePath.cd(ProjectBuildPathCmakeFiles);
+        }
+        return logFilePath.filePath(ProjectExportErrorLogName);
+    }
+
+    void ProjectExportWorker::QStringToAZTracePrint([[maybe_unused]] const QString& error)
+    {
+        AZ_Trace("Project Manager", error.toStdString().c_str());
+    }
+
+    AZ::Outcome<void, QString> ProjectExportWorker::ExportProjectForPlatform()
+    {
+        QString logFilePath = GetLogFilePath();
+        if(logFilePath == "")
+        {
+            QStringToAZTracePrint(LogPathFailureMsg);
+            return AZ::Failure(LogPathFailureMsg);
+        }
+        QFile logFile(logFilePath);
+        if (!logFile.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
+        {
+            QStringToAZTracePrint(LogOpenFailureMsg);
```

Comment:
There is a log file path here, instead of the fixed `LogOpenFailureMsg`, you should use a format string to also indicate the path of the log file in the error message.